File: jafu-beer-api/src/main/java/com/example/jafu/beer/api/controller/BeerHandler.java
package com.example.jafu.beer.api.controller;

import com.example.jafu.beer.api.entity.Order;
import com.example.jafu.beer.api.service.BarService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
public class BeerHandler {
    private final BarService barService;
    public Mono<ServerResponse> makeOrder(ServerRequest request) {
        return ServerResponse.status(HttpStatus.CREATED)
                .body(
                        request.bodyToMono(Order.class)
                                .map(barService::makeOrder),
                        String.class
                );
    }
}

File: kofu-beer-api/src/main/kotlin/com/example/kofu/beer/api/controller/BeerHandler.kt
package com.example.kofu.beer.api.controller

import com.example.kofu.beer.api.entity.Order
import com.example.kofu.beer.api.service.BarService
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.reactive.function.server.ServerRequest
import org.springframework.web.reactive.function.server.ServerResponse

class UserHandler(
        private val service: BarService
) {

    fun makeOrder(request: ServerRequest) = ServerResponse
            .status(HttpStatus.CREATED)
            .contentType(MediaType.TEXT_PLAIN)
            .body(request.bodyToMono(Order::class.java)
                    .map { order -> service.makeOrder(order) },
                    String::class.java
            )

}